tools/autograd/derivatives.yaml

```diff
@@ -208,7 +208,7 @@
 
 - name: add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> Tensor
   self: handle_r_to_c(self.scalar_type(), grad)
-  result: self_t
+  result: self_t.clone()
 
```

Comment:
maybe `auto_element_wise` will do the trick?

Reply:
I can see float to complex type promotion being something we might want here (I don't know how else people can reach this path without complex literals though, python binding already wraps scalar inputs with tensors):
```cpp
  using namespace c10::complex_literals;

  auto a = torch::ones({3, 3}, c10::TensorOptions().dtype(at::kFloat));
  auto b = a.add(0.5_id);
```
Not sure about `auto_element_wise` since we kind of want the opposite of  `handle_r_to_c` (cast to real) here right?